js/songs/new.controller.js

```diff
@@ -0,0 +1,16 @@
+"use strict";
+
+(function() {
+  angular
+  .module("songs")
+  .controller("newSongController", [
+    "songFactory",
+    songNewControllerFunction
+  ]);
+  function songNewControllerFunction(songFactory) {
+    this.song = new songFactory();
+    this.create = function() {
+      this.song.$save();
+    };
```

Comment:
This looks good, so don't think you're problem has to do with your `this.song` or `this.create` method definition.